far/editor.cpp

```diff
@@ -3970,6 +3999,41 @@ void Editor::DoSearchReplace(const SearchReplaceDisposition Disposition)
 			{ lng::MOk });
 }
 
+void Editor::SaveFoundItemsToNewEditor(const VMenu& ListBox, const bool MatchingFilter)
+{
+	const auto ShellEditor{ FileEditor::create(GetSearchAllFileName(), GetCodePage(), FFILEEDIT_CANNEWFILE | FFILEEDIT_ENABLEF6) };
+	const auto NewEditor{ ShellEditor->GetEditor() };
+	const auto FilterFlags{ LIF_HIDDEN | (MatchingFilter ? LIF_FILTERED : 0) };
+
+	using namespace std::views;
+
+	for (const auto I : ListBox.GetItems()
+		| filter([FilterFlags](const auto& Item) { return !(Item.Flags & FilterFlags); })
+		| transform([](const auto& Item) { return std::any_cast<FindCoord>(Item.ComplexUserData).Line; })
+		| chunk_by(std::ranges::equal_to{})
```

Comment:
`chunk_by` [requires](https://en.cppreference.com/w/cpp/compiler_support.html) gcc 13 / clang 18 / vs 2022 17.3, which are higher than our [current requirements](https://github.com/FarGroup/FarManager/blob/002d6c5e5f8a5f57ecff05a361cdcf55bcf0fa5d/far/headers.hpp#L40-L46).

Do we really need it here? Having `int LastLine = -1;` outside the loop and `if (I == LastLine) continue; LastLine = I;` inside it should achieve the same effect.

P.S. The requirements can probably be updated, but preferably not in this PR.

Reply:
Sure, I can do a simple loop. `chunk_by` here is a bit questionable performane-wise. I did not measure, but I have some intuition here.

What it takes to update requirements besides changing versions in the snippet you pointed? What is the highest we can reasonable jump to? I think keeping up with the latest compilers and language feature is a really good thing, not just a fad.

Reply:
> What it takes to update requirements besides changing versions in the snippet you pointed?

Make sure that our CI pipelines have those versions or newer.

> What is the highest we can reasonable jump to?

I think we discussed that already somewhere.
We do try to keep up with the latest, but also try to support 2-3 previous versions, released in the last 2-3 years or so.

Reply:
Thanks. I'll try to upgrade the compiler (separately of course). Lower priority for me. GH actions are defined in [`build.yml`](https://github.com/FarGroup/FarManager/blob/master/.github/workflows/build.yml), right? What about appveyor?